contrib/cindercompatibleapi/converter/volume.go

```diff
@@ -0,0 +1,380 @@
+	TenantID            string            `json:"os-vol-tenant-attr:tenant_id,omitempty"`
+	Migstat             string            `json:"os-vol-mig-status-attr:migstat,omitempty"`
+	Metadata            map[string]string `json:"metadata"`
+	Status              string            `json:"status"`
+	VolumeImageMetadata map[string]string `json:"volume_image_metadata,omitempty"`
+	Description         string            `json:"description"`
+	Multiattach         bool              `json:"multiattach,omitempty"`
+	SourceVolID         string            `json:"source_volid,omitempty"`
+	ConsistencygroupID  string            `json:"consistencygroup_id,omitempty"`
+	NameID              string            `json:"os-vol-mig-status-attr:name_id,omitempty"`
+	Name                string            `json:"name"`
+	Bootable            bool              `json:"bootable,omitempty"`
+	CreatedAt           string            `json:"created_at"`
+	VolumeType          string            `json:"volume_type,omitempty"`
+	ServiceUuID         string            `json:"service_uuid,omitempty"`
+	SharedTargets       bool              `json:"shared_targets,omitempty"`
+}
+
+// ShowVolumeResp ...
+func ShowVolumeResp(volume *model.VolumeSpec) *ShowVolumeRespSpec {
+	resp := ShowVolumeRespSpec{}
+
+	resp.Volume.Attachments = make([]RespAttachment, 0, 0)
+	resp.Volume.AvailabilityZone = volume.AvailabilityZone
+	resp.Volume.UpdatedAt = volume.BaseModel.UpdatedAt
+	resp.Volume.ID = volume.BaseModel.Id
+	resp.Volume.Size = volume.Size
+	resp.Volume.UserID = volume.UserId
+	resp.Volume.Metadata = make(map[string]string)
+	resp.Volume.Status = volume.Status
+	resp.Volume.Description = volume.Description
+	resp.Volume.Name = volume.Name
+	resp.Volume.CreatedAt = volume.BaseModel.CreatedAt
+	//resp.Volume.TenantID = volume.TenantId
+
+	return &resp
+}
+
+// *******************Update a volume*******************
+
+// UpdateVolumeReqSpec ...
+type UpdateVolumeReqSpec struct {
+	Volume UpdateReqVolume `json:"volume"`
+}
+
+// UpdateReqVolume ...
+type UpdateReqVolume struct {
+	Description string            `json:"description,omitempty"`
+	Name        string            `json:"name,omitempty"`
+	Metadata    map[string]string `json:"metadata,omitempty"`
+}
+
+// UpdateVolumeRespSpec ...
+type UpdateVolumeRespSpec struct {
+	Volume UpdateRespVolume `json:"volume"`
+}
+
+// UpdateRespVolume ...
+type UpdateRespVolume struct {
+	MigrationStatus    string            `json:"migration_status,omitempty"`
+	Attachments        []RespAttachment  `json:"attachments"`
+	Links              []Link            `json:"links,omitempty"`
+	AvailabilityZone   string            `json:"availability_zone,omitempty"`
+	Encrypted          bool              `json:"encrypted ,omitempty"`
+	UpdatedAt          string            `json:"updated_at"`
+	ReplicationStatus  string            `json:"replication_status,omitempty"`
+	SnapshotID         string            `json:"snapshot_id,omitempty"`
```

Comment:
the same

Reply:
In order to comply with golint's rules, "Id" must be written in all capitals.